engine/src/main/java/org/terasology/world/block/internal/BlockManagerImpl.java

```diff
@@ -93,7 +93,7 @@ public void initialise(List<String> registeredBlockFamilies,
         if (knownBlockMappings.size() >= MAX_ID) {
             nextId = UNKNOWN_ID;
         } else if (knownBlockMappings.size() > 0) {
-            nextId = (short) knownBlockMappings.size();
+            nextId = knownBlockMappings.values().stream().max(Short::compareTo).orElse((short) 0)+1;
```

Comment:
no, only once after loading a saved game. After initialisation only nextid has to be incremented.